layout/layers_ordering_assigner.go

```diff
@@ -282,35 +282,42 @@ func (o RandomLayerOrderingOptimizer) Optimize(segments map[[2]uint64]bool, laye
 	}
 }
 
-// time: O(ntop ^ 2 * nbot ^ 2)
-// memory: O(1)
+// time: O(ntop * nbot * log(ntop))
+// memory: O(ntop)
 func numCrossingsBetweenLayers(segments map[[2]uint64]bool, ltop, lbottom []uint64) int {
```

Comment:
perhaps keep both?

upside is that both solutions can test each other, and because they appear independent, this consensus will increase correctness guarantee 
```suggestion
func numCrossingsBetweenLayersBasic(segments map[[2]uint64]bool, ltop, lbottom []uint64) int {
```


```go
func numCrossingsBetweenLayersFenwick(segments map[[2]uint64]bool, ltop, lbottom []uint64) int {
~``